fw/rbcx-coprocessor/src/CdcUartTunnel.cpp

```diff
@@ -0,0 +1,99 @@
+/// Implements passthrough between the "primary" UART and USB CDC.
+
+#include "stm32f1xx_hal.h"
+#include "stm32f1xx_hal_dma.h"
+#include "stm32f1xx_ll_usart.h"
+
+#include "Bsp.hpp"
+#include "ByteFifo.hpp"
+#include "CdcUartTunnel.hpp"
+#include "UsbCdcLink.h"
+
+DMA_HandleTypeDef primaryUartDmaRxHandle;
+DMA_HandleTypeDef primaryUartDmaTxHandle;
+ByteFifo<512> primaryUartRxFifo;
+std::array<uint8_t, CDC_DATA_SZ> primaryUartTxBuf;
+
```

Comment:
Pokud jsou tyto proměnné používané pouze funkcemi z tohoto Cpp souboru (a nevidím teď protipříklad), zvážil bych jejich označení za static a/nebo jejich umístění do anonymního namespacu. Je to víceméně dobrá praktika, způsobí, že symbol nebude vidět mimo tuto kompilační jednotku a pokud si někdo ve svém cpp souboru pojmenuje symbol stejně, nedojde při linkování ke kolizi.

Zároveň z čitatelova pohledu to říká "Ok, vše, co potřebuji, je zde"

Reply:
Jj, o tomto dost často uvažuju a dospěl jsem tehdy k závěru, že míň kódu je čitelnější. Ale explicitnost je asi větší vejvar.

Preferujem `static` nebo `namespace {}` ?

Reply:
Zvládnu žít i bez toho.

Dříve jsem používal namespace, ale pak jsem začal dávat na rady Tonyho, že namespace přehlédneš, static ne.

Reply:
Já bych dál žil bez toho. Jiná věc by byla knihovna

Reply:
Já bych osobně preferoval být spíše explicitní. Ano, má to trochu side-effect v čitelnosti, ale předcházíš případným problémům.

Přijde mi to stejné jako uvádění `const` u proměnných.

Reply:
Const je totiž stejně blbej návrh jazyka C jako toto. Místo abys musel psát víc, když chceš víc, tak musíš psát víš, když chceš míň ehm Rust

Reply:
S tím naprosto souhlasím, bohužel ale používáme tento "blbý" jazyk, tak se tomu musíme přizpůsobit a psát víc a explicitněji, ne? Což by mělo vést k bezpečnějšímu kódu (jak `const` tak `static`).

Reply:
je to jak říkáš

Reply:
Provedeno 32bcad6